File: Practice/src/coding_basic_training/SumTermsOfArithmeticSequence.js
function solution(a, d, included) {
    var answer = 0;
    for(let i in included){
        if(included[i]) answer += a + d * i;
    }
    return answer;
    // reduce 버전
    // return included.reduce((ac, c, i) => (c ? ac + a + d * i : ac), 0);
}

File: Practice/src/coding_basic_training/SumTermsOfArithmeticSequence.java
package coding_basic_training;

public class SumTermsOfArithmeticSequence {
//	등차수열의 특정한 항만 더하기
	public int solution(int a, int d, boolean[] included) {
		int answer = 0;
		for (int i = 0; i < included.length; i++) {
			if (included[i])
				answer += a + d * i;
		}
		return answer;
//		intstream을 이용한 방법
//		return IntStream.range(0, included.length).map(idx -> included[idx] ? a + (idx * d) : 0).sum();
	}
}
